src/excmds.ts

```diff
@@ -2664,7 +2666,191 @@ export async function viewcontainers() {
+        await tabopen(initialUrl)
+        promises.push(tgroupRun(name, true).then(tabs => browserBg.tabs.hide(tabs.map(tab => tab.id))))
+    } else {
+        promises.push(browser.tabs.query({currentWindow: true}).then(async (tabs) => {
+            const ids = tabs.map(tab => tab.id)
+            await setTabTgroup(name, ids)
+        }))
+        promises.push(setWindowTgroup(name))
+    }
+
+    groups.add(name)
+    promises.push(setTgroups(groups))
+    return Promise.all(promises).then(() => undefined)
+}
+
+/**
+ * Switch to a different tab group, hiding all other tabs.
+ *
+ * @param name The name of the tab group to switch to.
+ *
+ * If the tab group does not exist, act like tgroupcreate.
+ *
+ */
+//#background
+export async function tgroupswitch(name: string) {
+    if (name == await windowTgroup()) {
+        throw new Error(`Already on tab group "${name}"`)
+    }
+
+    const groups = await tgroups()
+    if (groups.size > 0) {
+        if (groups.has(name)) {
+            return tgroupActivate(name).then(() => undefined)
+        } else {
+            return tgroupcreate(name).then(() => undefined)
+        }
+    } else {
+        return tgroupcreate(name).then(() => undefined)
+    }
+}
+
+/**
+ * Switch to the previously active tab group.
+ */
+//#background
+export async function tgrouplast() {
+    if ((await tgroups()).size < 2) {
+        throw new Error("No last tab group")
+    }
+
+    return tgroupActivateLast().then(() => undefined)
+}
+
+/**
+ * Rename the current tab group.
+ *
+ * @param name The new name of the tab group.
+ *
+ */
+//#background
+export async function tgrouprename(name: string) {
+    if ((await tgroups()).size == 0) {
+        throw new Error("No tab groups exist")
+    }
+
+    return tgroupClearOldInfo(await windowTgroup() as string, name).then(() => undefined)
+}
+
+/**
+ * Close the current tab group.
+ *
+ * First switch to the previously active tab group. Do nothing if there is only
+ * one tab group.
+ *
+ */
+//#background
+export async function tgroupclose() {
+    const groups = await tgroups()
+    if (groups.size == 0) {
+        throw new Error("No tab groups exist")
+    } else if (groups.size == 1) {
+        throw new Error("This is the only tab group")
+    } else if (groups.size > 1) {
+        const closeGroup = await windowTgroup() as string
+        await tgroupActivateLast()
+        await tgroupRun(closeGroup).then(tabs => browser.tabs.remove(tabs.map(tab => tab.id)))
+        return tgroupClearOldInfo(closeGroup).then(() => undefined)
+    }
+}
+
+/**
+ * Move the current tab to another tab group.
+ *
+ * @param name The name of the tab group to move the tab to.
+ *
+ * If this is the last tab in the tab group, also switch to tab group, keeping
+ * the current tab active.
+ *
+ */
+//#background
+export async function tgroupmove(name: string) {
+    const groups = await tgroups()
+    const currentGroup = await windowTgroup() as string
```

Comment:
```suggestion
    const currentGroup = await windowTgroup()
```
If you move the cast to windowTgroup.